io_scene_niftools/modules/nif_export/geometry/mesh/__init__.py

```diff
@@ -349,11 +366,17 @@ def export_tri_shapes(self, b_obj, n_parent, trishape_name=None):
             # update tangent space (as binary extra data only for Oblivion)
             # for extra shader texture games, only export it if those textures are actually exported
             # (civ4 seems to be consistent with not using tangent space on non shadered nifs)
-            if mesh_uv_layers and mesh_hasnormals:
-                if bpy.context.scene.niftools_scene.game in ('OBLIVION', 'FALLOUT_3', 'SKYRIM') or (bpy.context.scene.niftools_scene.game in self.texture_helper.USED_EXTRA_SHADER_TEXTURES):
-                    if bpy.context.scene.niftools_scene.game == 'SKYRIM':
-                        tridata.bs_num_uv_sets = tridata.bs_num_uv_sets + 4096
-                    trishape.update_tangent_space(as_extra=(bpy.context.scene.niftools_scene.game == 'OBLIVION'))
+            if use_tangents:
+                if bpy.context.scene.niftools_scene.game == 'SKYRIM':
+                    tridata.bs_num_uv_sets = tridata.bs_num_uv_sets + 4096
+                # calculate the bitangents using the normals, tangent list and bitangent sign
+                bitlist = bitangent_sign_list * np.cross(normlist, tanlist)
+                # B_tan: +d(B_u), B_bit: +d(B_v) and N_tan: +d(N_v), N_bit: +d(N_u)
+                # moreover, N_v = 1 - B_v, so d(B_v) = - d(N_v), therefore N_tan = -B_bit and N_bit = B_tan
+                self.add_defined_tangents(trishape,
+                                          tangents=-bitlist,
```

Comment:
Do you mean to have a `-` char here?

Reply:
Yes. As outlined in the comments above, the code, the bitangents in Nif format point towards positive nif V coordinate. In Blender, the tangents poit towards positive V coordinate. However, Nif V = 1- Blender V, so the direction is reversed.